helm/cluster-openstack/files/etc/patches/kube-apiserver-patch.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+#
+# Creates kube-apiserver+json.json file by replacing
+# environment variables in kube-apiserver+json.tpl
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cpus="$(grep -c ^processor /proc/cpuinfo)"
+memory="$(awk '/MemTotal/ { printf "%d \n", $2/1024 }' /proc/meminfo)"
+
+export MAX_REQUESTS_INFLIGHT=$((cpus*200))
+export MAX_MUTATING_REQUESTS_INFLIGHT=$((cpus*100))
+export API_SERVER_CPU_REQUEST=$((cpus*125))m
+export API_SERVER_MEMORY_REQUEST=$((memory / 8))Mi
```

Comment:
I tried to stay on the safe side but it might be too conservative. 

I thought this scenario: 2CPU/4GB flavor + MC. After removing the 2GB footprint of daemonsets and operating system, we have less than 2 GB. If I give 1/4 of total memory to api-server, the node will have less than 1GB for all other things. However, after rechecking everything again, I think this scenario is not a valid one. The 2GB footprint of daemonsets is for a highly loaded MC. If it is the case, we shouldn't use 2CPU/4GB flavor.  

On the other hand, it is `request` not `limit` so api-server can consume more. Now, I am questioning whether we should set `etcd` requests too now to protect `etcd` too.

Reply:
I think there is no single ratio that fits all cases. I added a parameter.